### PR TITLE
Generate XML documentation file to provide IntelliSense for consumers

### LINKED ISSUE #111: Display IntelliSense for consumers of NBuilder
As a consumer of this package, I would like IntelliSense displayed when I am using this package in my software so that I can better understand what the methods I am using are doing.

Generate the XML documentation file and package that into the NuGet package.

### PR DESCRIPTION
**Pull request by Doug-Murphy**
Also fully documents the GetRandom static class and fixes a few more minor issues for upper-exclusivity.

Resolves #111

### CODE REVIEW

**Comment by Doug-Murphy** [comment]
This _might_ require a change to the `New-NugetPackage.ps1` script. I'm not really sure though, so I'll defer to someone with knowledge on the build/release pipeline(s). `git log` is showing that @crmckenzie might be the person who would be familiar with that file.

File: src/FizzWare.NBuilder/Fizzware.NBuilder.csproj [hunk 1]
```diff
@@ -27,6 +27,14 @@
   <PropertyGroup Condition=" '$(TargetFramework)' == 'net40'">
     <DefineConstants>NET40;NETFULL</DefineConstants>
   </PropertyGroup>
+  
+  <PropertyGroup Condition="'$(Configuration)'=='Debug'">
+    <DocumentationFile>FizzWare.NBuilder.xml</DocumentationFile>
+  </PropertyGroup>
+  
+  <PropertyGroup Condition="'$(Configuration)'=='Release'">
```
Comment (PureKrome) [inline src/FizzWare.NBuilder/Fizzware.NBuilder.csproj]: Why include a Condition check for the same file? just replace lines 30 -> 37 with 

```
<PropertyGroup>
    <DocumentationFile>FizzWare.NBuilder.xml</DocumentationFile>
  </PropertyGroup>
```

so it's included regardless of the condition. Or is there something i'm miss understanding, here?

Comment (Doug-Murphy) [reply src/FizzWare.NBuilder/Fizzware.NBuilder.csproj]: This was auto-gen'd from when I set the documentation file via the project properties (not csproj directly). I just pushed an update to consolidate this!

**Comment by PureKrome** [comment]
Looks good!

Side (personal) NitPic: when writing git comments, maybe use the format:

```
First line: No `*` . This is the title line. max 72 odd chars.
2nd+ lines: start with a `*` or a `-` to generate a list of stuff, done. Or just plan text in a long line/paragraph.
```
Reasons: People can't easily see the commit history because GH doesn't like `*` in the title?

 
image

**Comment by Doug-Murphy** [comment]
Yeah it's a carryover from Git commits at work. We put the JIRA ticket on first line, then all subsequent lines are preceded with * to denote content of the commit.

I'll look at doing a more GitHub-friendly style for the future.

**Comment by phatcher** [comment]
Just an FYI, for SDK style projects you can use
`    <GenerateDocumentationFile>true</GenerateDocumentationFile>`
to automatically name the documentation file.  This doesn't work from the UI, just via editing the project file directly.

**Comment by Doug-Murphy** [comment]
@phatcher Thanks for the tip! If this repo was more active, I'd investigate your comment further and maybe make the change. However, since the PR doesn't look like it's going to get merged anyway, I'll forego it. =/

**Comment by PureKrome** [comment]
@garethdown44 ping!

**Comment by crmckenzie** [comment]
To include the docs in the package it's a 2-part process.

```xml
  <PropertyGroup>
    <GenerateDocumentationFile>true</GenerateDocumentationFile>
    <DocumentationFile>bin\$(Configuration)\$(TargetFramework)\$(AssemblyName).xml</DocumentationFile>
  </PropertyGroup>
```

As soon as github gets around to enabling the merge button for me, I'll merge it and add the appropriate xml to the project file.

**Comment by crmckenzie** [comment]
Since github won't let me merge this through the site, I manually merged it.
Thank you!

**Pull request closed without merging**